server/core/redis/redis.go

```diff
@@ -66,49 +66,136 @@ func NewWithClient(client *redis.Client, bucket string, globalBucket string) (*R
 // acquired, it will return true and the lock returned will be newLock.
 // If the lock is not acquired, it will return false and the current
 // lock that is preventing this lock from being acquired.
-func (r *RedisDB) TryLock(newLock models.ProjectLock) (bool, models.ProjectLock, error) {
+func (r *RedisDB) TryLock(newLock models.ProjectLock) (bool, models.ProjectLock, models.EnqueueStatus, error) {
 	var currLock models.ProjectLock
 	key := r.lockKey(newLock.Project, newLock.Workspace)
 	newLockSerialized, _ := json.Marshal(newLock)
+	var enqueueStatus models.EnqueueStatus // Lock Queue is not supported for Redis at the time being
 
 	val, err := r.client.Get(ctx, key).Result()
 	// if there is no run at that key then we're free to create the lock
 	if err == redis.Nil {
 		err := r.client.Set(ctx, key, newLockSerialized, 0).Err()
 		if err != nil {
-			return false, currLock, errors.Wrap(err, "db transaction failed")
+			return false, currLock, enqueueStatus, errors.Wrap(err, "db transaction failed")
 		}
-		return true, newLock, nil
+		return true, newLock, enqueueStatus, nil
 	} else if err != nil {
 		// otherwise the lock fails, return to caller the run that's holding the lock
-		return false, currLock, errors.Wrap(err, "db transaction failed")
+		return false, currLock, enqueueStatus, errors.Wrap(err, "db transaction failed")
 	} else {
 		if err := json.Unmarshal([]byte(val), &currLock); err != nil {
-			return false, currLock, errors.Wrap(err, "failed to deserialize current lock")
+			return false, currLock, enqueueStatus, errors.Wrap(err, "failed to deserialize current lock")
 		}
-		return false, currLock, nil
+		// checking if current lock is with the same PR
+		if currLock.Pull.Num == newLock.Pull.Num {
```

Comment:
Please break down additions to this function into smaller pieces for more unit tests

Reply:
Done as pat of 6937973d90fb5c94a400a1783e8e679966916cf4